### PR TITLE
Handle malformed X-Forwarded-For Header

### PR DESCRIPTION
**Pull request by bakgerman**
At the firm I work for, the network engineers have not followed X-Forwarded-For HHTP headers standards.
Instead of having the desired client IP at the start of the list of IP addresses in the header, the client IP is the second or third element, depending upon whether you are in development, test, or production network.

One way to solve this is to add an int Index property to the layout renderer.

### CODE REVIEW

File: src/Shared/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs [hunk 1]
```diff
@@ -35,6 +37,15 @@ public class AspNetRequestIpLayoutRenderer : AspNetLayoutRendererBase
...
         public bool CheckForwardedForHeader { get; set; }
 
+        /// <summary>
+        /// Gets or sets the array index of the X-Forwarded-For header to use, if the desired client IP is not at
+        /// the zeroth index.  Defaults to zero.  If the index is too large the last array element is returned instead.
+        /// If a negative index is used, this is used as the position from the end of the array.
+        /// Minus one will indicate the last element in the array.  If the negative index is too large the first index
+        /// of the array is returned instead.
+        /// </summary>
+        public int Index { get; set; } = 0;
```
Comment (snakefoot) [inline src/Shared/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs:47]: With all the nice contributions that you have made, then I'm very eager to help you out.

But I think `Index` is a little too generic for such an exotic option.

Could it be called `CheckForwardedForHeaderOffset` ? And be implemented like this (where activates offset +  header-lookup at the same time):

```c#
public int CheckForwardedForHeaderOffset
{
     get => _checkForwardedForHeaderOffset;
     set
     {
           _checkForwardedForHeaderOffset = value;
           CheckForwardedForHeader = true;
     }
}
```
And can be used like this:
```xml
${aspnet-request-ip:CheckForwardedForHeaderOffset=1}
```

Comment (bakgerman) [reply src/Shared/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs:47]: great idea, i will do that.  And thank you very much for all the help you provide me.

Comment (bakgerman) [reply src/Shared/LayoutRenderers/AspNetRequestIpLayoutRenderer.cs]: Completed

**Comment by snakefoot** [comment]
NLog.Web.AspNetCore v5.2.2 has been released - https://www.nuget.org/packages/NLog.Web.AspNetCore

**Pull request merged**